src/duckling/project/custom-attributes.component.ts

```diff
@@ -83,26 +84,23 @@ export class CustomAttributesComponent {
         this.curAddingName = newCurAddingName;
     }
 
-    onJsonSchemaChanged(newJsonSchema : JsonSchema) {
+    onJsonSchemaChanged(newJsonSchema : JsonSchemaEdit) {
         this.curAddingSchema = newJsonSchema;
     }
     
     onAcceptCustomAttribute() {
-        this._project.addCustomAttribute(this.curAddingName, schemaToJson(this.curAddingSchema));
-        this.curAddingName = "";
-        this.curAddingSchema = {keys: [], contents: []};
+        this._project.addCustomAttribute(this.curAddingName, schemaEditToJson(this.curAddingSchema));
+        this.curAddingName = undefined;
+        this.curAddingSchema = undefined;
```

Comment:
This change is incorrect. You need to set this to something other than undefined.